pkg/reloader/collector.go

```diff
@@ -93,24 +94,43 @@ func (c *Controller) collectWorkloadSecrets(workload workload, template corev1.P
 	collectorLogger.Infof("Collected secrets from %s %s/%s", workload.kind, workload.namespace, workload.name)
 }
 
-func collectSecretsFromContainerEnvVars(containers []corev1.Container) []string {
+func collectSecrets(template corev1.PodTemplateSpec) []string {
+	containers := []corev1.Container{}
+	containers = append(containers, template.Spec.Containers...)
+	containers = append(containers, template.Spec.InitContainers...)
+
+	vaultSecretPaths := []string{}
+	collectSecretsFromContainerEnvVars(containers, &vaultSecretPaths)
+	collectSecretsFromVaultEnvFromPathAnnotation(template.GetAnnotations()[SecretPathsAnnotation], &vaultSecretPaths)
```

Comment:
The two collect functions should return a slice which are then appended to `vaultSecretPaths` rather than passing the slice down by pointer and appending there